packages/react/src/components/Storage/File-Uploader/Previewer/index.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/** File Previewer */
+
+import React, { useRef, useState } from 'react';
+import { Button, Flex, Card, View } from '../../../../primitives';
+import { Tracker } from '../Tracker';
+import { FilePreviewerProps } from '../types';
+import { UploadTask } from '@aws-amplify/storage';
+
+export function Previewer({
+  fileNames,
+  level,
+  files,
+  onClose,
+}: FilePreviewerProps): JSX.Element {
+  const [percentage, setPercentage] = useState<number[]>([]);
+  const [uploadTasks, setUploadTasks] = useState<UploadTask[]>([]);
+  const [pauses, setPauses] = useState<boolean[]>([]);
```

Comment:
Wondering if we can combine them into a single array: 

```
interface FileStatus {
  percentage: number;
  uploadTask: UploadTask;
  pause: boolean
}

type FileStatuses = FileStatus[];

const [fileStatuses, setFileStatuses] = useState<FileStatuses>;
```
so we don't have to maintain an index for each array. I'm open to thoughts here though.

Reply:
I'll look at this in my next PR when I work on the Previewer, for now I'll delete